Rename Indexer client to Generator client throughout documentation (#253)
shinzonetwork/docs: content/views/overview/index.md

## Purpose and Role in the Stack

> Within the Shinzo ecosystem, View Creator sits above the Generator and alongside Hosts:

Reviewer: ```suggestion
Within the Shinzo ecosystem, View Creator sits above the Generator client and alongside Host clients:
```

> - Generator → produces normalized, deterministic primitive data

Reviewer: ```suggestion
- Generator clients → produces normalized, deterministic primitive data
```

- View Creator → defines how that data should be queried, transformed, and exposed

> - Host → executes those definitions, serves results, and attests to correctness

Reviewer: ```suggestion
- Host clients → executes those definitions, serves results, and attests to correctness
```
